mesos-simulation/src/test/scala/mesosphere/mesos/scale/MetricsResult.scala

```diff
@@ -0,0 +1,160 @@
+    *  { "a" : {}, "b" : {} } into an array of form
+    *  [ { "name": "a", ... }, { "name": "b", ...} ]
+    *  The given reads function is used to read the transformed json into an object.
+    */
+  def objectRead[T](t: Reads[T]): Reads[Seq[T]] = new Reads[Seq[T]] {
+    override def reads(js: JsValue): JsResult[Seq[T]] = {
+      JsSuccess(js.as[JsObject].fields.map {
+        case (name, value) =>
+          val obj = JsObject(value.as[JsObject].fields :+ ("name" -> JsString(name)))
+          t.reads(obj).get
+      })
+    }
+  }
+
+  // Json Reads objects for metric structure
+  implicit val gaugeReads = objectRead(Json.reads[Gauge])
+  implicit val countReads = objectRead(Json.reads[Counter])
+  implicit val histogramReads = objectRead(Json.reads[Histogram])
+  implicit val meterReads = objectRead(Json.reads[Meter])
+  implicit val timerReads = objectRead(Json.reads[Timer])
+  implicit val sampleReads: Reads[MetricsSample] = Json.reads[MetricsSample]
+
+  def readMetrics(file: File): Seq[MetricsSample] = {
+    val jsonString = FileUtils.readFileToString(file)
+    Json.parse(jsonString).as[Seq[MetricsSample]]
+  }
+
+  /**
+    * Compare 2 Metrics Samples and filter all metrics, that get deteriorated by given factor.
+    * @param baseLine the base line to compare the values
+    * @param sample the metric samples to compare
+    * @param factor the deterioration factor. 1 means not worse than the base.
+    * @return all deteriorated metrics (before, after)
+    */
+  def filterDeteriorated(baseLine: MetricsSample, sample: MetricsSample, factor: Double): Map[Metric, Metric] = {
+    for {
+      (name, metricsBase) <- baseLine.all
+      metricsSample <- sample.all.get(name).toList
+      metricBase <- metricsBase if metricBase.mean > 0
+      metricSample <- metricsSample.find(_.name == metricBase.name)
+      if (metricSample.mean / metricBase.mean) > factor
+    } yield metricBase -> metricSample
+  }
+
+  def filterDeteriorated(beforeFile: File, afterFile: File, deterioration: Double): Map[Metric, Metric] = {
+    //only compare last
+    filterDeteriorated(readMetrics(beforeFile).last, readMetrics(afterFile).last, deterioration)
+  }
+
+  def main(args: Array[String]) {
+    if (args.length == 3) {
+      val deteriorated = filterDeteriorated(new File(args(0)), new File(args(1)), args(2).toDouble)
+      if (deteriorated.nonEmpty) {
+        val message = deteriorated.map{
+          case (b, a) =>
+            s"${b.name}: ${b.mean} -> ${a.mean} [${(a.mean / b.mean * 100).toInt}%]"
+        }.mkString("\n")
+        throw new IllegalStateException(message)
+      }
+    }
+    else {
+      println(
+        """Usage:
+          | MetricsResult <file_base> <file_sample> <deterioration_factor>"
+          | file_base: the file with the base metrics
+          | file_sample: the file with the actual sampled metrics
+          | deterioration_factor: number [0..] where 1 means the same as the base line
```

Comment:
Ah, ok. This is the allowed regression. So the each metric is allowed to be greater than the base line by this factor. A factor smaller than 1 is probably not really useful.

Even in the good case, it would be useful to print metrics which have become worse (including percentages) sorted descendingly by their factor.

I also like to align the information. You could use the same classes I wrote for DisplayAppScalingResults (DisplayHelpers.printTable).